--- images.py.orig	2013-09-26 07:39:01.905862277 -0600
+++ images.py	2013-09-25 08:54:12.984328984 -0600
@@ -55,7 +55,7 @@
 LOG = logging.getLogger(__name__)
 SUPPORTED_PARAMS = glance.api.v1.SUPPORTED_PARAMS
 SUPPORTED_FILTERS = glance.api.v1.SUPPORTED_FILTERS
-CONTAINER_FORMATS = ['ami', 'ari', 'aki', 'bare', 'ovf']
+CONTAINER_FORMATS = ['ami', 'ari', 'aki', 'bare', 'ovf', 'docker']
 DISK_FORMATS = ['ami', 'ari', 'aki', 'vhd', 'vmdk', 'raw', 'qcow2', 'vdi',
                 'iso']
 
